pkg/cmd/build/authprovider.go

```diff
@@ -148,36 +148,47 @@ func (ap *authProvider) Credentials(ctx context.Context, req *auth.CredentialsRe
 		return nil, err
 	}
 
-	credentials := ap.getOktetoCredentials(originalHost, c)
+	retrieveFromLocalFirst := env.LoadBooleanOrDefault(oktetoLocalRegistryStorePriorityEnabledEnvVarKey, false)
+	oktetoCredentials := ap.getOktetoCredentials(originalHost, c)
+	localCredentials, err := ap.getLocalCredentials(req)
+	if err != nil {
+		return nil, err
+	}
 
-	retrieveFromLocal := env.LoadBooleanOrDefault(oktetoLocalRegistryStoreEnabledEnvVarKey, false)
-	if !retrieveFromLocal {
-		return credentials, nil
+	areCredentialsValid := func(creds *auth.CredentialsResponse) bool {
+		return creds != nil && creds.Username != "" && creds.Secret != ""
 	}
 
+	// Return local credentials if prioritizing local and valid, otherwise Okteto credentials if valid, fallback to local
+	if (retrieveFromLocalFirst && areCredentialsValid(localCredentials)) || !areCredentialsValid(oktetoCredentials) {
```

Comment:
Why do we check now if `oktetoCredentials` are valid? Before we were not checking it and we were defaulting to the Okteto ones

Reply:
We were not checking it because we were checking if the localCredentials were valid and if not we returned the oktetoCredentials. We now have to give priority and need to check both of them